pkg/cluster/connection_pooler.go

```diff
@@ -284,7 +291,7 @@ func (c *Cluster) generateConnectionPoolerPodTemplate(role PostgresRole) (
 
 	podTemplate := &v1.PodTemplateSpec{
 		ObjectMeta: metav1.ObjectMeta{
-			Labels:      c.connectionPoolerLabelsSelector(role).MatchLabels,
+			Labels:      c.connectionPoolerLabels(role, false).MatchLabels,
```

Comment:
question here is: Does it have to be the same list of what's in the selector? Or has the selector only be contained in the pod template label list (for watching)